bin/reth/src/args/network_args.rs

```diff
@@ -122,6 +122,19 @@ impl NetworkArgs {
 
         Some(peers_file)
     }
+
+    /// Sets the p2p port to zero, to allow the OS to assign a random unused port when
+    /// the network components bind to a socket.
+    pub fn with_unused_p2p_port(&mut self) {
+        self.port = 0;
+    }
+
+    /// Sets the p2p and discovery ports to zero, allowing the OD to assign a random unused port
+    /// when network components bind to sockets.
+    pub fn with_unused_ports(&mut self) {
```

Comment:
do we want &mut here or self?

we can have two functions if convenient, though the &mut self should also return &mut Self

Reply:
changed to `mut self -> Self`